src/lock/parse.rs

```diff
@@ -15,38 +15,230 @@ impl Lock {
                 details: e.details,
             })?;
             dependencies.push(dep);
+pub fn parse(text: &str) -> Result<Vec<LockedDependency>, Error> {
+    let mut res = vec![];
+    let lines = split_logical_lines(&text);
+    for line in lines.iter() {
+        if !line.starts_with('#') {
```

Comment:
This could be a filter/map if you want to avoid the mut Vec, but that's okay (yes, you can collect an iter of `Results` into a `Result<Vec<_>>`!)

Reply:
Ah. Good to know. I prefer the loop so that I can explain why I'm skipping the comment.

And it would coming in handy if we need to implement the span later on